internal/wasmldr/src/loader/requested.rs

```diff
@@ -0,0 +1,157 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use super::{pki::PrivateKeyInfoExt, Attested, Loader, Requested};
+
+use std::time::Duration;
+use std::{io::Read, ops::Deref, sync::Arc};
+
+use anyhow::{anyhow, Result};
+
+use pkcs8::PrivateKeyInfo;
+use rustls::{cipher_suite::*, kx_group::*, version::TLS13, *};
+use url::Url;
+use x509::der::asn1::{BitString, UIntBytes};
+use x509::der::{Decodable, Encodable};
+use x509::ext::pkix::{BasicConstraints, ExtendedKeyUsage, KeyUsage, KeyUsages};
+use x509::name::RdnSequence;
+use x509::time::Validity;
+use x509::{Certificate, PkiPath, TbsCertificate};
+
+use const_oid::db::rfc5280::{
+    ID_CE_BASIC_CONSTRAINTS, ID_CE_EXT_KEY_USAGE, ID_CE_KEY_USAGE, ID_KP_CLIENT_AUTH,
+    ID_KP_SERVER_AUTH,
+};
+
+impl Loader<Requested> {
+    fn steward(&self, url: &Url) -> Result<Vec<Vec<u8>>> {
+        if url.scheme() != "https" {
+            return Err(anyhow!("refusing to use an unencrypted steward url"));
+        }
+
+        // Send the attestation to the steward.
+        let response = ureq::post(url.as_str())
+            .set("Content-Type", "application/pkcs10")
+            .send_bytes(&self.0.crtreq)?;
+
+        // Read the result.
+        let mut body = Vec::new();
+        response.into_reader().read_to_end(&mut body)?;
+
+        // Decode the certificate chain.
+        let path = PkiPath::from_der(&body)?;
+        path.0
+            .iter()
+            .rev()
+            .map(|c| Ok(c.to_vec()?))
+            .collect::<Result<Vec<_>, _>>()
```

Comment:
Is this turbofish necessary? I would expect it to be inferred from the return type.

Reply:
Removed.